cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -376,3 +757,32 @@ func (s *Service) deleteInstanceGroups(ctx context.Context) error {
 
 	return nil
 }
+
+// getSubnet gets the subnet to use for an internal Load Balancer.
+func (s *Service) getSubnet(ctx context.Context) (*compute.Subnetwork, error) {
+	log := log.FromContext(ctx)
+	cfgSubnet := ""
+	lbSpec := s.scope.LoadBalancer()
+	if lbSpec.InternalLoadBalancer != nil {
+		cfgSubnet = ptr.Deref(lbSpec.InternalLoadBalancer.Subnet, "")
+	}
+	for _, subnetSpec := range s.scope.SubnetSpecs() {
+		log.V(2).Info("Looking for subnet for load balancer", "name", subnetSpec.Name)
+		region := subnetSpec.Region
+		if region == "" {
+			region = s.scope.Region()
+		}
+
+		subnetKey := meta.RegionalKey(subnetSpec.Name, region)
+		subnet, err := s.subnets.Get(ctx, subnetKey)
+		if err != nil {
+			return nil, err
+		}
+		// Return subnet that matches configuration or first one if not configured
+		if cfgSubnet == "" || strings.HasSuffix(subnet.Name, cfgSubnet) {
```

Comment:
What is the reason for using HasSuffix instead of a full match?

Reply:
The full name includes the cluster name so it just matches the subnet portion. I can update this to build the name including cluster and subnet for comparision.

Reply:
Sounds good